packages/block-library/src/navigation/edit/index.js

```diff
@@ -310,6 +325,15 @@ function Navigation( {
 	] = useState();
 	const [ detectedOverlayColor, setDetectedOverlayColor ] = useState();
 
+	useEffect( () => {
+		if (
+			! classicMenuConversionState?.isFetching &&
+			classicMenuConversionState.navMenu
+		) {
+			setRef( classicMenuConversionState.navMenu?.id );
```

Comment:
This is confusing. Due to the `useState` var name containing the term `ref` It could appear we are setting a `React.ref` here which would not cause the component to re-render and thus the UI would not reflect the new menu ID.

In actual fact this is a `setState` call.

We should look to refactor this `ref` term away to something else.